app/models/index-card.ts

```diff
@@ -44,6 +68,39 @@ export default class IndexCardModel extends Model {
         }
         return '';
     }
+
+    @dropTask
+    @waitFor
+    async getOsfModel() {
+        const identifier = this.resourceIdentifier;
+        if (identifier && this.osfModelType) {
+            const guid = this.guidFromIdentifierList(identifier);
+            const relatedCounts = this.osfModelType === 'user' ? 'nodes,registrations,preprints' : '';
+            if (guid) {
+                const osfModel = await this.store.findRecord(this.osfModelType, guid, {
+                    adapterOptions: {
+                        query: {
+                            related_counts: relatedCounts,
+                        },
+                    },
+                    reload: true,
+                });
+                this.osfModel = osfModel;
+            }
+        }
+    }
+
+    guidFromIdentifierList(ids: string[]) {
```

Comment:
since this is a method on the model, would make sense for it to look at `this.resourceIdentifier` instead of taking an `ids` argument

Reply:
Good point! I didn't update it much after moving it into this model